### PR TITLE
Gsoc: drop six dependencies

### PR DESCRIPTION
**Pull request by unkcpz**
@sphuber 
This PR 
- remove all `six` dependency and 
- remove all `from __future__` clause. 
- remove < 3.5 test in .travis ci, add an additional python 3.8 test

I assume to bring pre-commit related modified task in the next parallel PR, so that all subsequent changes can have code style checks easily turned on.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -279,7 +279,7 @@ init-import=no
 
 # List of qualified module names which can have objects that can redefine
 # builtins.
-redefining-builtins-modules=six.moves,future.builtins
+redefining-builtins-modules=future.builtins
```
Comment (sphuber) [inline .pylintrc:282]: Can't the `future.builtins` also not be removed? Or does that have nothing to do with the `__future__` statements?

File: setup.py [hunk 2]
```diff
@@ -37,7 +36,6 @@
           'kiwipy[rmq]>=0.5.2',
           'enum34; python_version<"3.4"',
```
Comment (sphuber) [inline setup.py:37]: we can also remove the `enum34` dependency now

Comment (sphuber) [reply setup.py:37]: And the `backports.tempfile` as well

File: test/base/test_statemachine.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 PAUSED = 'Paused'
 STOPPED = 'Stopped'
 
+# TODO: test exceptions
```
Comment (sphuber) [inline test/base/test_statemachine.py:16]: remove please

File: .pylintrc (hunk 1 above)
Comment (unkcpz) [reply .pylintrc]: Hh, forget to remove it, cause I did not take care of __future__ at previous PR.

File: setup.py (hunk 2 above)
Comment (unkcpz) [reply setup.py]: done.

File: test/base/test_statemachine.py (hunk 3 above)
Comment (unkcpz) [reply test/base/test_statemachine.py]: erased.

**Comment by unkcpz** [comment]
Except for `_tb_logger` which still used by tornado Future not removed, all you suggested are done. 
`_tb_logger` will be removed after tornado.concurrent.Future is replaced.

**Comment by unkcpz** [comment]
And I turn on pre-commit this time, so there are some code style fixes.

**Comment by sphuber** [comment]
Thanks a lot @unkcpz ! Great first step. I will now first move from Travis to Github actions just like all our other projects and then we can continue with the rest of your changes

**Pull request merged**